script_parameters.yml

```diff
@@ -0,0 +1,238 @@
+# This example workflow demonstrates how to pass parameters into different types
+# of scripts:
+#   1) SQL script
+#   2) Container script
+#   3) Python script
+#   4) R script
+#   5) Custom script
+#
+# It generally involves adding a `params` list and an `arguments` hash to the
+# task inputs.  The `params` list defines the names and types of the parameters
+# that the script can take.  The `arguments` hash contains the values for those
+# parameters.  Think of it like a form, where `params` defines the checkboxes
+# and input fields on the form, and `arguments` is filling the form out.
+#
+# Currently, it is not possible to pass parameters from a workflow into an
+# existing script job (using the `civis.run_job` action).  Passing parameters
+# into a script will create a new job for each execution of the workflow.
+#
+# Inside your script, parameter values are made available via templating (for
+# SQL scripts) or via environment variables (for Container, Python, and R
+# scripts).  See the Script parameter reference linked below for a reference
+# containing all the available parameter types, and what variables are made
+# available to your script.  If you are passing in string parameters from user
+# input or other untrusted sources, we advise using escaped versions of the
+# variables to prevent injection attacks.
+#
+# Links:
+# * Workflows documentation (requires Platform login): https://civis.zendesk.com/hc/en-us/articles/115004172983-Workflows-Basics
+# * Script parameter reference (requires Platform login): https://civis.zendesk.com/hc/en-us/articles/360001579592-Configuring-Scripts-via-the-Civis-API
+# * Mistral domain specific language reference: https://docs.openstack.org/mistral/latest/user/wf_lang_v2.html
+# * YAML syntax: https://docs.ansible.com/ansible/latest/reference_appendices/YAMLSyntax.html
+
+version: '2.0'  # you always need this key to specify version 2 of the mistral DSL
+workflow:
+  tasks:
+    sql_script:
+      action: civis.scripts.sql
+      input:
+        # SQL scripts always require the IDs of the remote host and the credential
+        # you are using to connect to it.
+        remote_host_id: 0  # get it from https://api.civisanalytics.com/databases
+        credential_id: 0   # get it from https://platform.civisanalytics.com/credentials
+        params:  # This is not an exhaustive list of param types - see the script parameter reference for more
+          - name: my_integer
+            type: integer
+          - name: my_float
+            type: float
+          - name: my_string
+            type: string
+          - name: my_custom_credential
+            type: credential_custom
+        arguments:
+          my_integer: 1234
+          my_float: 0.1234
+          my_string: this is a string
+          my_custom_credential: 0  # ID of the credential
+        sql: |
+          SELECT 1;
```

Comment:
Just an observation here that the other examples give some code example of using calling the parameter into the script type. Something to consider would be adding a simple, common query example. Maybe something like "SELECT * FROM {{table_name}} WHERE {{col_name}} > {{my_integer}} ?

Reply:
I was thinking about that, but then I thought maybe people would copy-and-paste the YAML into a workflow on Platform, and would want to be able to run it and see parameters in action without modifying the script code.  Maybe we can have it both ways by keeping the `SELECT 1` and adding the query example to the commented section?

Reply:
I think that makes sense!